precompile/contract/interfaces.go

```diff
@@ -51,19 +51,24 @@ type AccessibleState interface {
 	GetSnowContext() *snow.Context
 }
 
-// BlockContext defines an interface that provides information to a stateful precompile about the
-// current block. The BlockContext may be provided during both precompile activation and execution.
-type BlockContext interface {
+// ConfigurationBlockContext defines the interface required to configure a precompile.
+type ConfigurationBlockContext interface {
 	Number() *big.Int
 	Timestamp() uint64
 }
 
+// BlockContext defines the interface to provide information about the block to a precompile
+// during the execution of the EVM.
+type BlockContext interface {
+	ConfigurationBlockContext
+}
+
```

Comment:
As @darioush said should we separate these changes for another pr and let this one focus on the predicates